### PR TITLE
added buffer and worker limit

### PR DESCRIPTION
**Pull request by facundomedica**

### CODE REVIEW

**Comment by almogdepaz** [comment]
looks very good!
the base intention was that users of msgserver can pass a buffer of the selected size
so i want msgserver to receive the size in NewMsgServer.
you can put the hardcoded value in the call to NewMsgServer and we will fix this later on.

File: p2p/server/msgserver.go [hunk 1]
```diff
@@ -49,10 +50,11 @@ func NewMsgServer(network Service, name string, requestLifetime time.Duration, l
 		resHandlers:        make(map[uint64]func(msg []byte)),
 		pendingQueue:       list.New(),
 		network:            network,
-		ingressChannel:     network.RegisterProtocol(name),
+		ingressChannel:     network.RegisterProtocolWithChannel(name, make(chan service.Message, 100)),
```
Comment (almogdepaz) [inline p2p/server/msgserver.go]: change to NewMsgServer(network Service, name string, requestLifetime time.Duration,c chan service.Message ,logger log.Log)
and pass c to RegisterProtocolWithChannel

Comment (facundomedica) [reply p2p/server/msgserver.go]: And what about workerLimiter? Do I pass an int or just leave it hardcoded?

Comment (facundomedica) [reply p2p/server/msgserver.go]: And what about workerLimiter? Do I pass an int or just leave it hardcoded?

Comment (almogdepaz) [reply p2p/server/msgserver.go]: you can leave the worker limit as it is

**Comment by facundomedica** [comment]
@almogdepaz it should be done now

File: sync/syncer.go [hunk 2]
```diff
@@ -107,7 +108,7 @@ func NewSync(srv server.Service, layers *mesh.Mesh, bv BlockValidator, conf Conf
 		Log:            logger,
 		Mesh:           layers,
 		Peers:          p2p.NewPeers(srv),
-		MessageServer:  server.NewMsgServer(srv, syncProtocol, conf.requestTimeout-time.Millisecond*30, logger),
+		MessageServer:  server.NewMsgServer(srv, syncProtocol, conf.requestTimeout-time.Millisecond*30, make(chan service.Message, 100), logger),
```
Comment (almogdepaz) [inline sync/syncer.go]: put the hardcoded 100 in the config

File: p2p/server/msgserver.go (hunk 1 above)
Comment (almogdepaz) [reply p2p/server/msgserver.go]: actually would be better to use runtime.numcpu() for the worker limit 
you can hardcode it

**Comment by facundomedica** [comment]
@almogdepaz some unrelated tests are failing, but this should be done

**Comment by almogdepaz** [comment]
@facundomedica i reran your tests just rebase and ill approve

**Comment by facundomedica** [comment]
@almogdepaz oops, I'm not good at rebasing. Let me try fixing it

**Pull request closed without merging**